pkg/store/volume.go

```diff
@@ -0,0 +1,69 @@
+package store
+
+import (
+	"errors"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"github.com/lima-vm/lima/pkg/qemu/imgutil"
+	"github.com/lima-vm/lima/pkg/store/filenames"
+)
+
+type Volume struct {
+	Name        string `json:"name"`
+	Size        int64  `json:"size"`
+	Dir         string `json:"dir"`
+	DiskPath    string `json:"diskPath"`
+	Instance    string `json:"instance"`
+	InstanceDir string `json:"instanceDir"`
+}
+
+func InspectVolume(volName string) (*Volume, error) {
+	vol := &Volume{
+		Name: volName,
+	}
+
+	volDir, err := VolumeDir(volName)
+	if err != nil {
+		return nil, err
+	}
+
+	vol.Dir = volDir
+	dataDisk := filepath.Join(volDir, filenames.DataDisk)
+	if _, err := os.Stat(dataDisk); err != nil {
+		return nil, err
+	}
+	vol.DiskPath = dataDisk
+
+	info, err := imgutil.GetInfo(vol.DiskPath)
+	if err != nil {
+		return nil, err
+	}
+	vol.Size = info.VSize
+
+	instDir, err := os.Readlink(filepath.Join(volDir, filenames.InUseBy))
+	if err != nil {
+		if errors.Is(err, fs.ErrNotExist) {
+			vol.Instance = ""
+			vol.InstanceDir = ""
+		} else {
+			return nil, err
+		}
+	} else {
+		vol.Instance = filepath.Base(instDir)
+		vol.InstanceDir = instDir
+	}
+
+	return vol, nil
+}
+
+func (v *Volume) Lock(instanceDir string) error {
+	inUseBy := filepath.Join(v.Dir, filenames.InUseBy)
+	return os.Symlink(instanceDir, inUseBy)
```

Comment:
Shouldn't we use flock?

Reply:
We could. `flock` is just advisory, but since this is only checked by lima, that doesn't matter. The advantage of the symlink is that the `basename(symlink)` gives us the instance name, so we can see which instance holds the lock.